### DISCUSSION TITLE
Running Scalene with elevated privileges does not produce profiler results.

### DISCUSSION THREAD

**Discussion by MrDrDoom** [plasma-umass/scalene#430] [Q&A] [answered]
Hello,

I am using Ubuntu 20.04 LTS on a [HardKernel ODroid M1 board](https://www.hardkernel.com/shop/odroid-m1-with-8gbyte-ram/), and I am using the terminal to launch Scalene/my application.

The application I am trying to profile needs elevated access in order to open and maintain serial connection as well as write to logging files. If I run the command `sudo scalene main.py`, my app starts up and operates almost as intended but does not open up my browser to show the profiling results. If I run the command `scalene main.py`, the app fails to start up as it needs elevated access for the components I mentioned before but it does launch my browser and offers the little it was able to profile.

Is there a way to run Scalene with elevated privilege that I missed in the documentation?

Thank you in advance for any help.

**Comment by emeryberger** [comment, marked as answer]
So I assume you are running your program normally with `sudo python3 main.py`, correct?

I installed Scalene as root (not recommended), but it worked. You might try this, which I verified works for me on Ubuntu.

`sudo bash -c "scalene --cli ./main.py"`
